http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskServiceInterp.scala

```diff
@@ -230,6 +230,8 @@ class DiskServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
             )
             .toVector
       }
+      // If no visible disks, check if user should be able to use this endpoint at all
+      _ <- if (res.isEmpty) authProvider.checkUserEnabled(userInfo) else F.unit
```

Comment:
question, why are we checking if the user should be able to use the endpoint after the rest of the logic is executed? Would it not make more sense to do this check initially and give the error response if the user isn't enabled before trying to fetch info from the DB?

Reply:
This was my feeling initially, but the idea was to only do this check if no resources were visible. Could move this to the top but if so I'd want to just do it without a conditional - apply the second call to all requests.